app/controllers/oauth_clients_controller.rb

```diff
@@ -219,9 +219,12 @@ def find_oauth_client
 
   # rubocop:enable Metrics/AbcSize
 
+  def oauth_integration
+    @oauth_client&.integration
+  end
+
   def redirect_user_or_admin(redirect_uri = nil)
-    # This needs to be modified as soon as we support more integration types.
-    if User.current.admin && redirect_uri && (nextcloud? || one_drive? || share_point?)
+    if User.current.admin && redirect_uri && oauth_integration.try(:supports_oauth_redirect?)
```

Comment:
:yellow_circle: Do we need some specifics of the `try` method added by ActiveSupport or would the "modern" safe navigation operator be sufficient?

```suggestion
    if User.current.admin && redirect_uri && oauth_integration&.supports_oauth_redirect?
```

Reply:
I've started with the `&.` at the beginning, but then I've noticed that `integration` is a polymorphic connection
```
class OAuthClient < ApplicationRecord
  belongs_to :integration, polymorphic: true
```

it seems, as if now, that only storage has this connection to the model, but in future, other models could use it and they would not have `supports_oauth_redirect?` method. 

so I've continued with:
```
oauth_integration.respond_to?(:supports_oauth_redirect?) && oauth_integration.supports_oauth_redirect?
```

which eventually became just `try`.

Reply:
That's a fair difference. I usually prefer the specificness of `&.` and it's true: It would require that every implementor supports this method as well.

But I agree that your implementation is safer and will not raise errors.